### PR TITLE
WT-7592 Remove support for WT_SESSION::transaction_sync and background sync.

### PR DESCRIPTION
**Pull request by sueloverso**
@agorrod Here are the changes to remove support for background sync in log_flush and transactions, to give you an idea of what/how much code it removes. AFAIK no one has ever used this interface, but it is possible my knowledge is incomplete. We can either take this branch or close it and leave it in the code.

### CODE REVIEW

File: src/docs/upgrading.dox [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
 
+</dl><hr>
+@section version_1001 Upgrading to Version 10.0.1
+<dl>
+
+<dt>Background sync APIs</dt>
+<dd>
+The WT_SESSION::background_sync API and the sync=background configuration option to
+WT_SESSION::log_flush and WT_SESSION::commit_transaction have all been removed from WiredTiger.
+</dd>
```
Comment (keithbostic) [inline src/docs/upgrading.dox:11]: Should this be `WT_SESSION::transaction_sync`?

(FTR, I think the upgrading document is just going away, I should remove it and push that change so it's less confusing.)

**Review by keithbostic — APPROVED** [review]
The WT-7592 ticket doesn't mention WT_SESSION.transaction_sync.

This change seems reasonable to me, but because I don't know this area, I'm not likely to catch anything that was missed.

File: src/docs/upgrading.dox (hunk 1 above)
Comment (sueloverso) [reply src/docs/upgrading.dox:11]: Good catch. Fixed.

**Comment by sueloverso** [comment]
I agree it is difficult to catch anything else that should be removed. I counted on the compiler and scripts for most of that (and it caught the stats that were now not being used).

**Review by agorrod — APPROVED** [review]
The change here LGTM, pending confirmation from Storage Execution that the code isn't useful.

**Pull request merged**